### PR TITLE
medrxivr blogpost

### PR DESCRIPTION
**Pull request by mcguinlu**
Good morning @ropensci/blog-editors, 

Thanks for your patience, as I appreciate this is a day late! I had encountered issues when trying to previewing the post locally with Hugo, but I've checked it using the Netlify build and everything looks as it should.

As this is my second blogpost ever, very happy to incorporate any feedback you have!

Thanks for the opportunity to contribute to the blog,

Luke

### CODE REVIEW

**Comment by steffilazerte** [comment]
@mcguinlu No worries! I'll take a look at it today and get back to you soon. Thanks!

File: content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+One of the key assumptions of Boolean operators is that the order of the terms does not matter, i.e. that "(dementia OR alzheimer) AND lipids" is equivalent to "(alzheimer OR dementia) AND lipids". However, running these searches on the 13th Oct 2020, the first gives 133 results and the second gives 126. It is not clear what is happening behind the scenes, but there is potentially some weighting of terms based on proximity. This means that key records may be missed based on the arbitrary ordering of the search terms.
+
+<br>
+
+### Lack of reproducbility
+
+One of the key tenets of systematic reviews is that literature searches are transparent and reproducible - that is, that a certain search strategy (e.g. "(dementia OR alzheimer) AND lipids") will provide the same number of results each time it is performed. However, anecdotal evidence from information specialists and medical librarians suggests this is not the case for the medRxiv native search:
```
Comment (steffilazerte) [inline content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:50]:
```suggestion
One of the key tenets of systematic reviews is that literature searches are transparent and reproducible - that is, that a certain search strategy (e.g. "(dementia OR alzheimer) AND lipids") will provide the same number of results each time it is performed. However, anecdotal evidence from information specialists and medical librarians suggests that medRxiv's websearch returns inconsistent citations over time.
```

File: content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+<!--html_preserve-->
+{{< tweet 1244905414019428353 >}}
+<!--/html_preserve-->
+
+<!--html_preserve-->
+{{< tweet 1255485844783325185 >}}
+<!--/html_preserve-->
+
+This means that records returned by a search must be exported by hand in small batches, an error-prone and time-consuming (not to mention anger-inducing!) method of extracting citations from a bibliographic database.
```
Comment (steffilazerte) [inline content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:70]: Absolutely fine to say "anger-inducing" here :grin:, I'm sure it is!

**Review by steffilazerte — COMMENTED** [review]
Hi @mcguinlu 

You said this is your second blog post (congrats)! I really love how awesome the medrxivr package is. It's always great to see a problem solved so elegantly. Your use cases are so lovely and clean and really show case how this package can help so many researchers. I do love my job as an editor :)

In general, most of my comments are stylistic, I think you may have missed some of the check boxes on the [author submission list](https://blogguide.ropensci.org/authorchecklistpeer.html). But those are very quick fixes. The content is well organized, clear and easy to understand. And, as I mentioned above, the use cases are great.

The one thing I'm a bit uncomfortable with is some of the really frustrated tweets. Clearly the interface with medRxiv leaves something to be desired, but I don't want you or rOpenSci to look like you're just bashing medRxiv. I think you illustrate the drawbacks in the text clearly and with grace, without the need for the more negative tweets.

Thanks again for your well written post and useful package!

* [X] post follows Content Guidelines
* [x] post follows Style Guide
* [X] title is in Title Case
* [X] publication date is ok
* [X] alternative text of images is informative
* [X] Twitter metadata looks ok (paste post preview link in [Twitter card validator](https://cards-dev.twitter.com/validator); might have to click twice on "Preview card")
* [X] author metadata is provided with correct folder name
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on index.md
* [X] I ran `roblog::ro_check_urls()` on index.md
* [X] I ran a spell-check on index.md
* [X] YAML subject tags are ok
* [X] YAML package-version included
* [X] YAML subject tags - software peer review, packagename, R; add "community" for post by non-staff non-editor
* [x] acknowledges and links to reviewers
* [x] links to peer review thread

**Comment by steffilazerte** [comment]
Oh and if I could get your revisions by Friday or Monday, that'd be great!

File: content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  theme_classic() 
+```
+
+<!--html_preserve-->
+{{< figure src = "categories-1.png" width = "600" alt = "Distribution of preprints posted to medRxiv in the first week of 2020 across topic categories." >}}
+<!--/html_preserve-->
+
+### Use Case #2 – Perform transparent and reproducible searches as part of a literature review, and export results
+
+While being able to easily import medRxiv data is useful, as a systematic reviewer, my main interest is in what `medrxivr` allows you to do once you have imported it. As mentioned in the section above, the native medRxiv website search interface is neither reproducible nor transparent. The second core aspect of `medrxivr`'s functionality is designed to address this, allowing information specialists to build complex searches and find records contained in the medRxiv databases that match the criteria, all while documenting their search strategy. An example is given below, where a researcher is looking for records that contain both dementia-related and lipid-related terms:
```
[the remark this answers is not in the document]
Comment (mcguinlu) [reply content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:110]: So your second interpretation is correct - to search for relevant articles, you first have to import a local copy of the _whole_ database, which is then passed as the `data` argument to `mx_search()`.

I've added some text to this section to try and make this point clearer!

File: content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+```
+
+Once you have run your search, exporting the results to a .BIB file for import into a reference manager, such as Zotero or Mendeley, is a simply as passing the results object to the `mx_export()` function.
+
+Similarly, as screening through the full text PDFs of records returned by a search against eligibility criteria forms a key part of a systematic review, a second helper function downloads the PDF for each record returned by `mx_search()`. Again, it is simply a case of passing the `results` object to the `mx_download()` function.
+
+## Conclusion
+
+medRxiv is a fantastic resource and has been a key source of information related to the COVID-19 pandemic. However, some key issues with the website mean that its native search functionality precludes its use in systematic reviews. The `medrxivr` R package seeks to address these limitations by providing a user-friendly way to import and systematically search medRxiv and bioRxiv records in R. Full documentation on the functionality, particularly around the implementation of complex search strategies using syntax such as wildcards and NEAR, is available from the [package website](https://docs.ropensci.org/medrxivr/).
```
[the remark this answers is not in the document]
Comment (mcguinlu) [reply content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:135]: I've added some text to explain the NEAR operator, but please let me know if it is still unclear!

[the remark this answers is not in the document]
Comment (mcguinlu) [reply content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:135]: I've added a linkto the review thread, and added both reviewers names/links to GH profiles to the text.

**Comment by mcguinlu** [comment]
Hi @steffilazerte 👋

Thanks a million for your helpful comments - hopefully I have addressed them in my recent commits. I completely agree about the tweets - I just wanted to get across that it's not just me that has encountered these issues and that they are recognised problems in the community, but I can see in retrospect that they come across a bit strong! I've removed them now in favour of the text.

You are right to say I missed a few items on the checklist - in actual fact, I missed the checklist completely! 🤦‍♂️ Very sorry about that, but I think I have addressed all of the items on it now. Just to flag that `roblog::ro_check_url()` was not happy with some of the URLs in the post and I am not entirely sure why. . . . I've reproduced the output below, but just let me know if anything needs fixing:

``` 
Unknown option: verb
Unknown option: verb
Unknown option: verb
Unknown option: verb
Unknown option: verb
Unknown option: verb
Unknown option: verb
Unknown option: verb
Possibly broken URLs: https://www.medrxiv.org/, https://docs.ropensci.org/medrxivr/, https://www.biorxiv.org/, https://github.com/tts, https://github.com/njahn82, https://github.com/ropensci/software-review/issues/380, https://docs.ropensci.org/medrxivr/, https://doi.org/doi:10.1136/bmj.l2301.
```

**Comment by steffilazerte** [comment]
That's great! Thanks for your speedy edits. I'm a bit slow this week (got a cold, bah! With so much social distancing I thought that was impossible!) but I'll hopefully finish the review this afternoon.

File: content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+  theme_classic() 
+```
+
+<!--html_preserve-->
+{{< figure src = "categories-1.png" width = "600" alt = "Distribution of preprints posted to medRxiv in the first week of 2020 across topic categories." >}}
+<!--/html_preserve-->
+
+### Use case #2 – Reproducible searching and exporting
+
+While being able to easily import medRxiv data is useful, as a systematic reviewer, my main interest is in what medrxivr allows you to do once you have imported a local copy of the database. As mentioned in the section above, the native medRxiv website search interface is neither reproducible nor transparent, and while the API (accessed via `mx_api_content()`) is great for accessing medRxiv metadata, it does not have offer any search functionality. The second core aspect of medrxivr's functionality is designed to address these limitations, allowing information specialists to build complex searches and apply them to a local copy of the medRxiv database, all while documenting their search strategy in a transparent manner. An example is given below, where a researcher is looking for records that contain both dementia-related and lipid-related terms:
```
Comment (steffilazerte) [inline content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:94]:
```suggestion
While being able to easily import medRxiv data is useful, as a systematic reviewer, my main interest is in what medrxivr allows you to do once you have imported a local copy of the database. As mentioned in the section above, the native medRxiv website search interface is neither reproducible nor transparent, and while the API (accessed via `mx_api_content()`) is great for accessing medRxiv metadata, it does not offer any search functionality. The second core aspect of medrxivr's functionality is designed to address these limitations, allowing information specialists to build complex searches and apply them to a local copy of the medRxiv database, all while documenting their search strategy in a transparent manner. An example is given below, where a researcher is looking for records that contain both dementia-related and lipid-related terms:
```

File: content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+``` {r}
+
+# Use the maintained snapshot to quickly load today's copy of the medRxiv database
+# Note - this `preprint_data` object could be saved as a CSV in your project repository to aid reproducibility
```
Comment (steffilazerte) [inline content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:99]: Makes so much more sense!
Small edit to make it fit :smile:

```suggestion
# Note - this `preprint_data` object could be saved as a CSV in your project 
#        repository to aid reproducibility
```

File: content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+# Build your search
+topic1  <- c("dementia","vascular","alzheimer's")  # Combined with Boolean OR
+topic2  <- c("lipids","statins","cholesterol")     # Combined with Boolean OR
+myquery <- list(topic1, topic2)                    # Combined with Boolean AND
+
+# Run your search
+results <- mx_search(data = preprint_data,
+                     query = myquery)
+
```
Comment (steffilazerte) [inline content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:110]: Perhaps print the results to show the awesome tibble that is returned
```suggestion
results
```

File: content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+<!--html_preserve-->
+{{< figure src = "categories-1.png" width = "600" alt = "Distribution of preprints posted to medRxiv in the first week of 2020 across topic categories." >}}
+<!--/html_preserve-->
+
+### Use case #2 – Reproducible searching and exporting
+
+While being able to easily import medRxiv data is useful, as a systematic reviewer, my main interest is in what medrxivr allows you to do once you have imported a local copy of the database. As mentioned in the section above, the native medRxiv website search interface is neither reproducible nor transparent, and while the API (accessed via `mx_api_content()`) is great for accessing medRxiv metadata, it does not have offer any search functionality. The second core aspect of medrxivr's functionality is designed to address these limitations, allowing information specialists to build complex searches and apply them to a local copy of the medRxiv database, all while documenting their search strategy in a transparent manner. An example is given below, where a researcher is looking for records that contain both dementia-related and lipid-related terms:
+
+``` {r}
```
Comment (steffilazerte) [inline content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:96]: Below I suggest printing the results, if you like that idea, then I also suggest increasing the width, just a bit to show more of the table
```suggestion
``` {r, R.options=list(width=100)}
```

File: content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+## Conclusion
+
+medRxiv is a fantastic resource and has been a key source of information related to the COVID-19 pandemic. However, some key issues with the website mean that its native search functionality precludes its use in systematic reviews. The medrxivr R package seeks to address these limitations by providing a user-friendly way to import and systematically search medRxiv and bioRxiv records in R.  Full documentation of the package functionality, particularly around the implementation of complex search strategies using syntax such as wildcards and the NEAR operator, which is used to find co-located terms (e.g. "cholesterol NEAR2 test" find records where "cholesterol" and "test" are separated by two or less words), is available from the [medrxivr website](https://docs.ropensci.org/medrxivr/). Finally, I wanted to note that during development, medrxivr benefited immensely from an [rOpenSci peer review](https://github.com/ropensci/software-review/issues/380) by [Tuija Sonkkila](https://github.com/tts) and [Najko Jahn](https://github.com/njahn82).
```
Comment (steffilazerte) [inline content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:119]: I had no idea there was such thing as a NEAR operator. That is SO cool

**Review by steffilazerte — COMMENTED** [review]
This looks great! I have one typo fix and two suggestions for the final code block (feel free to take or leave them). 
Thanks for your 🚀  work

File: content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd (hunk 9 above)
Comment (mcguinlu) [reply content/blog/2020-10-20-searching-medrxivr-and-biorxiv-preprint-data/index.Rmd:119]: Yeah, you can currently use it in `medrxivr` via some pretty ugly regex syntax, but implementing it in a nicer fashion is definitely on the horizon!

**Comment by mcguinlu** [comment]
Great, thanks! Edits committed, plus one small final change from me (short comment in the code block just before the `results` object is printed). 

 Wishing you a very nice weekend, and a speedy recovery from the cold!

**Comment by steffilazerte** [comment]
Awesome, we're good to go! Thank you for your post, I'll publish and tweet on Tuesday 😸

**Comment by stefaniebutland** [comment]
@mcguinlu I just noticed that in the post, the package itself isn't mentioned until "Use cases". Would you consider adding a sentence or two right before Background so people know what the post and package are about? Maybe consider adding to the title? Searching medRxiv and bioRxiv Preprint Data with the medrxivr package.

cc @steffilazerte

**Comment by steffilazerte** [comment]
@mcguinlu That's right, I completely missed that you removed a paragraph from the beginning, definitely consider adding it back in, I quite liked it 😄 

```
However, the native search functionality available on the medRxiv website is not suitable for
use in systematic reviews, due to a number of limitations. These limitations, presented below,
prompted the development of [medrxivr](https://docs.ropensci.org/medrxivr) R package which 
provides access to, and tools for searching, medRxiv preprint metadata. This post will also
summarize the key functionality of the package with respect to two key use cases. Note that 
medrxivr allows users to access and search bioRxiv as well as medRxiv preprint metadata, 
as both repositories are run by the Cold Spring Harbor Laboratory and so have a similar 
API endpoint/native website search functionality, but all illustrating examples in this post are
taken from medRxiv.
```

**Comment by mcguinlu** [comment]
Whoops - not entirely sure how I managed to delete that paragraph, but have re-added it now! Great catch @stefaniebutland!

**Pull request merged**